tests/Unit/EmogrifierTest.php

```diff
@@ -3334,4 +3334,34 @@ public function emogrifyNotInDebugModeKeepsInvalidOrUnrecognizedSelectorsInMedia
 
         self::assertContainsCss($css, $result);
     }
+
+    /**
+     * @test
+     */
+    public function copyUninlineableCssToStyleNodeHasNoSideEffects()
+    {
+        $this->subject->setHtml('<html><a>foo</a><p>bar</p></html>');
+        $css = 'a:hover { color: green; } p:hover { color: blue; }';
+
+        $parseCssRules = new \ReflectionMethod(Emogrifier::class, 'parseCssRules');
+        $parseCssRules->setAccessible(true);
+        $copyUninlineableCssToStyleNode = new \ReflectionMethod(Emogrifier::class, 'copyUninlineableCssToStyleNode');
+        $copyUninlineableCssToStyleNode->setAccessible(true);
```

Comment:
I really don't like us accessing private methods as this will make refactoring a lot harder. Please let's use public methods only for the test.

Reply:
It's not easily possible to test the private method another way.  Perhaps we don't want to make this change permanent, but it would be useful to confirm that changes made for #668 don't result in the method having unexpected side-effects.

Reply:
Though looking again, the parameter `$cssRules['uninlineable']` could be set up as static data, so that only the method under test is accessed privately...

Reply:
I've now changed this so the data passed to the method under test is initialized as an array instead of by invoking private methods.